mica/starcheck/starcheck.py

```diff
@@ -26,52 +28,167 @@
              sql_def='starcheck.sql')
 
 # this doesn't belong in starcheck
-def get_timeline_at_date(date):
-    aca_db = Ska.DBI.DBI(dbi='sybase', server='sybase', user='aca_read')
-    return aca_db.fetchone(
+def get_timeline_at_date(date, timelines_db=None):
+    date = DateTime(date).date
+    if timelines_db is None:
+        timelines_db = Ska.DBI.DBI(dbi='sqlite',
+                         server='/proj/sot/ska/data/cmd_states/cmd_states.db3')
+    return timelines_db.fetchone(
         "select * from timeline_loads where datestop >= '%s' "
         " and datestart <= '%s' and scs <= 130 order by datestart desc"
         % (date, date))
 
-def get_mp_dir(obsid, config=None):
+
+def get_flickpix_mons(start=None, config=None):
     if config is None:
         config = DEFAULT_CONFIG
-    aca_db = Ska.DBI.DBI(dbi='sybase', server='sybase', user='aca_read')
-    possible_runs = aca_db.fetchall(
-        "select * from tl_obsids where obsid = %d" % obsid)
-    if not len(possible_runs):
-        db = Ska.DBI.DBI(dbi='sqlite', server=config['server'])
-        sc = db.fetchall(
+    db = Ska.DBI.DBI(dbi='sqlite', server=config['server'])
+    # I'm ignoring the weird case of SIR obsids or whatever where the following ER keeps
+    # the MON
+    mons = db.fetchall("""select obsid, mp_starcat_time, type, sz, yang, zang, dir
+                          from starcheck_catalog, starcheck_id
+                          where type = 'MON' and obsid > 40000
+                          and starcheck_id.id = starcheck_catalog.sc_id""")
+    mons = Table(mons)
+    if start is not None:
```

Comment:
OK, I was too lazy to thing through the conditional SQL, but your `start_date` is fine.  And not sure about your 'mp_starcat_time' comment.  It is actually a date, just has a that `time` because it has been around a while (before I was good about date vs time).  I could alias it, rename it in the database or whatever.

Unrelatedly, while I love the comments, I thought "WIP" was the thing to do if I wasn't ready for them.